UserGuide.md

```diff
@@ -260,6 +261,36 @@ gson.fromJson(json, fooType);
 ```
 The idiom used to get `fooType` actually defines an anonymous local inner class containing a method `getType()` that returns the fully parameterized type.
 
+### <a name="TOC-Parsing-a-Collection-with-Objects-of-Non-Arbitrary-Types"></a>Parsing a Collection with Objects of Non Arbitrary Types
```

Comment:
Maybe instead of "non arbitrary types" -> "specific types"?
However, this section only covers arrays, so maybe it would be better to add it to the existing section "Array Examples".